src/coreclr/vm/comdelegate.cpp

```diff
@@ -925,8 +925,7 @@ FCIMPL5(FC_BOOL_RET, COMDelegate::BindToMethodName,
 
     // get the name in UTF8 format
     SString wszName(SString::Literal, gc.methodName->GetBuffer());
-    StackScratchBuffer utf8Name;
-    LPCUTF8 szNameStr = wszName.GetUTF8(utf8Name);
+    LPCUTF8 szNameStr = wszName.GetUTF8();
```

Comment:
This will have an unmanaged heap allocation now that was not here before. 

There is probably at least one perf regression hidden in this PR from the extra copies or extra allocations.

Should we unify all these places that want to get SString with particular encoding to one efficient pattern that avoids heap allocations and extra copies?

Reply:
Given the myriad of states an `SString` can be in, I don't know if there's a single best way to transcode without an additional allocation. The "literal" state in particular makes this quite difficult.

This is why I liked the idea of encoding some of these concepts in the type system, so we could write more consistent patterns more easily.

Reply:
I looked through the changes in this commit and this change here is the only one where we are introducing a heap allocation that wasn't already there.

I'll still add the `SetAndConvertToUTF8` method as it will allow us to get rid of the `GetUTF8` method that takes the scratch buffer for a few more cases.